lib/rules/no-nested-landmark.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const Rule = require('./base');
+const AstNodeInfo = require('../helpers/ast-node-info');
+
+function createErrorMessage(element) {
+  return `Nested landmark elements on <${element}> detected. Landmark elements should not be nested within landmark elements of the same name.`;
+}
+
+const LANDMARK_ELEMENTS = ['header', 'main', 'aside', 'form', 'main', 'nav', 'footer'];
+const ROLES = ['banner', 'main', 'complementary', 'form', 'search', 'navigation', 'contentinfo'];
```

Comment:
Can you add an inline comment just above each of these lines to link to the spec where they are defined?